src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -558,15 +569,40 @@ private[fastq] object FastqDemultiplexer {
   /** Stores the minimal information for a single template read. */
   case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
                          sampleBarcode: Option[String], readNumber: Int, pairedEnd: Boolean, comment: Option[String],
-                         originalBases: Option[String] = None, originalQuals: Option[String] = None)
+                         originalBases: Option[String] = None, originalQuals: Option[String] = None) {
+
+    /** A function to mask bases that have a quality score less than or equal to a specified threshold
+      * @param threshold The threshold for masking bases, inclusive. Bases with a quality score <= threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: Char, qualityEncoding: QualityEncoding): DemuxRecord = {
+      if (this.quals.exists(_ <= threshold)) {
+        val bases = this.bases.toCharArray
+        for (i <- Range(0, this.quals.length)) if (quals.charAt(i) <= threshold) bases(i) = 'N'
+        this.copy(bases = bases.mkString)
+      }
+      else this
+    }
+  }
 
   /** A class to store the [[SampleInfo]] and associated demultiplexed [[DemuxRecord]]s.
     * @param sampleInfo the [[SampleInfo]] for the matched sample.
     * @param numMismatches the # of mismatches if it matched a sample with a sample barcode.  This will be [[Int.MaxValue]]
     *                      for the unmatched sample.
     * @param records the records, one for each read that has template bases.
     */
-  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord])
+  case class DemuxResult(sampleInfo: SampleInfo, numMismatches: Int, records: Seq[DemuxRecord]) {
+
+    /** A function to mask bases that have a quality score less than or equal to a specified threshold
+      * @param threshold The threshold for masking bases, inclusive. Bases with a quality score <= threshold will be masked
+      * @return a new DemuxResult with updated bases
+      */
+    def maskLowQualityBases(threshold: Char, qualityEncoding: QualityEncoding): DemuxResult = { // using this.type here causes a mismatch error
```

Comment:
I think we have a type alias you can use wherever you have `Char` for a PhredScore.  Can you fix here and everywhere else?
```suggestion
    def maskLowQualityBases(threshold: PhredScore, qualityEncoding: QualityEncoding): DemuxResult = { // using this.type here causes a mismatch error
```